<?xml version="1.0"?>
<launch>

  <!-- CAN card S/N -->
  <arg name="hardware_id" default="1" />
  <!-- Channel index on the card -->
  <arg name="circuit_id" default="0" />
  <!-- CAN bitrate / CANFd arbitration phase bitrate -->
  <arg name="bitrate_0" default="500" />
  <!-- CANFd data phase bitrate -->
  <arg name="bitrate_1" default="2000" />
  <!-- CAN accept filter code -->
  <arg name="filter_code" default="0" />
  <!-- CAN accept filter mask -->
  <arg name="filter_mask" default="0" />
  <!-- CAN slider window size -->
  <arg name="window_size" default="4" />

  <!-- is CAN FD ? true = 1 -->
  <arg name="is_canfd" default="1" />
  <!-- is Exclusive -->
  <arg name="is_exclusive" default="0" />
  <!-- is Support Virtual -->
  <arg name="is_virtual" default="1" />

  <node pkg="kvaser_can_rospkg" type="kvasercan_bridge" name="kvaser_can_bridge" output="screen">
    <param name="can_hardware_id" value="$(arg hardware_id)" />
    <param name="can_circuit_id" value="$(arg circuit_id)" />
    <param name="can_bitrate_0" value="$(arg bitrate_0)" />
    <param name="can_bitrate_1" value="$(arg bitrate_1)" />
    <param name="can_fliter_code" value="$(arg filter_code)" />
    <param name="can_filter_mask" value="$(arg filter_mask)" />
    <param name="can_window_size" value="$(arg window_size)" />
    <param name="can_is_canfd" value="$(arg is_canfd)" />
    <param name="can_is_exclusive" value="$(arg is_exclusive)" />
    <param name="can_is_virtual" value="$(arg is_virtual)" />
  </node>

</launch>
  


